app/scripts/views/mixins/password-strength-mixin.js

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'lib/promise',
+  'lib/runtime-require'
+], function (p, runtimeRequire) {
+  'use strict';
+
+  var PasswordStrengthMixin = {
+    _getPasswordStrengthChecker: function () {
+      // returns a promise that resolves once the library is loaded.
+      var self = this;
+      if (! self._passwordStrengthCheckerPromise) {
+        self._passwordStrengthCheckerPromise = runtimeRequire('passwordcheck')
+          .then(function (PasswordCheck) {
+            return new PasswordCheck();
+          });
+      }
+      return self._passwordStrengthCheckerPromise;
+    },
+    /**
+     * Check the password strength. Returns a promise that resolves
+     * when the check is complete.
+     *
+     * Usage:
+     *
+     *   view.checkPasswordStrength(password)
+     *     .then(function (status) {
+     *      // do something with the status
+     *     });
+     *
+     * @method checkPasswordStrength
+     * @returns {Promise}
+     */
+    checkPasswordStrength: function (password) {
+      var self = this;
+      return this._getPasswordStrengthChecker()
+        .then(function (passwordStrengthChecker) {
+          var deferred = p.defer();
```

Comment:
@vladikoff I love the idea of having both Y and Z, and later removing the ones that arent useful enough.
I think the reason we added those remaining three, was for completeness. If we are anyway checking for those, then why not gather metrics on them as well?
Eg: If `PASSWORD_TOO_SHORT` occurs x% of the time, maybe we should better highlight the `Password must be of minimum 8 characters` message.
If `MISSING_PASSWORD` occurs frequently, we can see if this happens due to the focus being lost on the password field before the user has a chance to type in(maybe correlate this with the type of device?)
`PASSWORD_NOT_A_STRING` *almost* always points to an attack vector. Only someone who can override jquery's default .val() method can make this happen. Might be useful to check if someones trying to attack us.